Newsletters: add 220 (2022-10-05) (#874)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2022-10-05-newsletter.md

---
title: 'Bitcoin Optech Newsletter #220'
permalink: /en/newsletters/2022/10/05/
name: 2022-10-05-newsletter
slug: 2022-10-05-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter FIXME

## News

- **Proposed new transaction relay policies designed for LN-penalty:**
  Gloria Zhao [posted][] to the Bitcoin-Dev mailing list a proposal to
  allow transactions to opt-in to a modified set of transaction relay
  policies.  Any transaction which sets its version parameter to `3`
  will:

    - Be replacable while it is unconfirmed by a transaction paying a
      higher feerate and a higher total fee (the current main
      [RBF][topic rbf] rules)

    - Be required to set its version parameter to `3` and follow all of
      these rules if any of the transactions it depends on (its ancestor
      transactions) are both unconfirmed and also version `3`

>     - Be rejected if any of its unconfirmed ancestors already have any

Reviewer: ```suggestion
    - Be rejected if any of its unconfirmed V3 ancestors already have any
```

I know it's draft, but wanted to correct this. Non-V3 transactions' descendant rules are still mostly respected. I do apologize as I know the rules are pretty convoluted/hacky.

To illustrate what I mean (where <- means "spends," i.e. parent <- child. "large" means >1000vB):
![image](https://user-images.githubusercontent.com/25183001/193531846-01e46645-62b7-4e07-82a6-9c61e87387a1.png)

      relay] containing this transaction)

    - Be required to be 1,000 vbytes or smaller if any of its version
      `3` ancestors are unconfirmed

    Accompaning the proposed relay rules was a simplification of the
    previously-proposed package relay relay rules (see [Newsletter
    #FIXME][]).

    Together, the v3 relay and updated package relay rules are designed
    to allow LN commitment transactions to include only minimal fees (or
    potentially even zero fees) and have their actual fees paid by a
    child transaction, all while preventing [pinning][].  Almost all LN
    nodes already use a mechanism like this, [anchor outputs][], but the
    proposed upgrade should provide make confirming commitment
    transactions simpler and more robust.

    Greg Sanders [replied][] with two suggestions:

    - *Emphemeral dust:* any transactions paying a zero-value (or
      otherwise *uneconomical* amount) output should be exempt from the
      [dust policy][] if that transaction is part of a package which
      spends the dust output.  

    - *Standard OP_TRUE:* that outputs paying an output consisting
      entirely of `OP_TRUE` should be relayed by default.  Such an

>       output can be spend by anyone---it has no security.  That makes it

Reviewer: ```suggestion
      output can be spent by anyone---it has no security.  That makes it
```

      easy for either party to an LN channel (or even third parties) to
      fee-bump a transaction spending that `OP_TRUE` output.  No data
      needs to be put on the stack to spend an `OP_TRUE` output, making

>       it efficient to spend.

Reviewer: ```suggestion
      it cost-efficient to spend.
```

---
title: 'Bitcoin Optech Newsletter #220'
permalink: /en/newsletters/2022/10/05/
name: 2022-10-05-newsletter
slug: 2022-10-05-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter FIXME

## News

- **Proposed new transaction relay policies designed for LN-penalty:**
  Gloria Zhao [posted][] to the Bitcoin-Dev mailing list a proposal to
  allow transactions to opt-in to a modified set of transaction relay
  policies.  Any transaction which sets its version parameter to `3`

>   will:

Author: That's fair.  I'm going to reprase the second bullet point rather than all the rest as I think we want signal to readers that we're only summarizing the v3 rules.

    nodes already use a mechanism like this, [anchor outputs][topic anchor outputs], but the
    proposed upgrade should make confirming commitment
    transactions simpler and more robust.

    Greg Sanders [replied][sanders tx3] with two suggestions:

    - *Ephemeral dust:* any transactions paying a zero-value (or
      otherwise *uneconomical*) output should be exempt from the
      [dust policy][topic uneconomical outputs] if that transaction is part of a package which
      spends the dust output.

    - *Standard OP_TRUE:* that outputs paying an output consisting
      entirely of `OP_TRUE` should be relayed by default.  Such an
      output can be spent by anyone---it has no security.  That makes it
      easy for either party to an LN channel (or even third parties) to
      fee-bump a transaction spending that `OP_TRUE` output.  No data
      needs to be put on the stack to spend an `OP_TRUE` output, making
      it cost-efficient to spend.

    Neither of these needs to be done at the same time as implementing
    relay of v3 transactions, but several respondents to the thread

>     seemed to be in favor of all the proposed changes.

Reviewer: V3 summary looks correct to me 👍 thanks @harding!

---
title: 'Bitcoin Optech Newsletter #220'
permalink: /en/newsletters/2022/10/05/
name: 2022-10-05-newsletter
slug: 2022-10-05-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter describes a proposal for new opt-in transaction relay
rules and summarizes research into helping LN channels stay balanced.
Also included are our regular sections listing new software releases and
release candidates plus notable changes to popular Bitcoin infrastructure
projects.

## News

- **Proposed new transaction relay policies designed for LN-penalty:**
  Gloria Zhao [posted][zhao tx3] to the Bitcoin-Dev mailing list a proposal to
  allow transactions to opt-in to a modified set of transaction relay
  policies.  Any transaction which sets its version parameter to `3`
  will:

    - Be replaceable while it is unconfirmed by a transaction paying a
      higher feerate and a higher total fee (the current main
      [RBF][topic rbf] rules)

    - Be rejected if any of the transactions it depends on (its ancestor

>       transactions) are both unconfirmed and also v3

Reviewer 2: I'm confused by the image in an earlier comment conjuction with the current text. It seems that all of the “valid” examples in the image would be in conflict with this rule. Is it possible that there was a mixup here?

Reviewer: Thanks @Xekyo! Looks like fixed in 7166a56

    - Be rejected if any of its v3 unconfirmed ancestors already have
      any other descendants in the mempool (or in a [package][topic
      package relay] containing this transaction)

    - Be required to be 1,000 vbytes or smaller if any of its v3

>       ancestors are unconfirmed

Reviewer 2: Isn't “Be rejected if any of the transactions it depends on (its ancestor transactions) are both unconfirmed and also v3” and “Be required to be 1,000 vbytes or smaller if any of its v3 ancestors are unconfirmed” contradictory? Why would the second statement matter if they're generally rejected already when they have unconfirmed v3 ancestors?

Author: Yeah, this was the result of an attempted change.  Fixing.

[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
Proposals (BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*

- [Eclair #2435][] adds optional support for a basic form of *async
  payments* when [trampoline relay][topic trampoline payments] is used.  As described in
  [Newsletter #171][news171 async], async payments would allow paying an
  offline node (such as a mobile wallet) without trusting a third-party
  with the funds.  The ideal mechanism for async payments depends on
  [PTLCs][topic ptlc], but a partial implementation just requires a third party to
  delay forwarding the funds until the offline node comes back online.
  Trampoline nodes can provide that delay and so this PR makes use of
  them to allow experimentation with async payments.

- [BOLTs #962][] removes support for the original fixed-length onion
  data format from the specification.  The upgraded variable-length
  format was added to the specification over three years ago and test

>   results mentioned in the commit message indicate almost no one is
>   using the older format.

Reviewer 2: ```suggestion
  results mentioned in the commit message indicate almost no one is
  using the older format anymore.
```

- [BIPs #1370][] revises [BIP330][] ([Erlay][topic erlay] for reconciliation-based
  transaction announcements) to reflect the current proposed
  implementation. Changes include:

  - Removing truncated transaction IDs in favor of just using
    transaction wtxids.  This also means nodes can use the existing
    `inv` and `getdata` messages, so the `invtx` and `gettx` messages
    have been removed.

  - Renaming `sendrecon` to `sendtxrcncl`,
  `reqreconcil` to `reqrecon`, and `reqbisec` to `reqsketchtext`.

  - Adding details for negotiating support using `sendtxrcncl`.

- [BIPs #1367][] simplifies [BIP118][]'s description of
  [SIGHASH_ANYPREVOUT][topic sighash_anyprevout] by refering to BIPs [340][bip340] and
  [341][bip341] as much as possible.

- [BIPs #1349][] adds BIP351 titled “Private Payments”. Alfred Hodler
  and Clark Moody describe a cryptographic protocol inspired by
  [BIP47][bip47] and [Silent Payments][topic silent payments]. The BIP
  introduces a new Payment Code format per which participants specify
  supported output types next to their public key.  Similar to BIP47, a
  sender uses a notification transaction to establish a shared secret

>   with the receiver from the receiver's Payment Code. The sender can

Reviewer 2: ```suggestion
  with the receiver based on the receiver's Payment Code. The sender can
```

  then send multiple payments to unique addresses derived from the
  shared secret which the receiver may spend using information from the
  notification transaction. Where BIP47 had multiple senders reuse the
  same notification address per receiver, this proposal uses OP_RETURN

>   outputs labeled with the search key `PP` and a receiver-specific
>   notification code to get the receiver's attention and establish the

Reviewer 2: ```suggestion
  outputs labeled with the search key `PP` and a
  notification code specific to the sender-receiver pair to get the receiver's attention and establish the
```

  shared secret for improved privacy.

- [BIPs #1293][] adds [BIP372][] titled "Pay-to-contract tweak fields for PSBT". This BIP
  proposes a standard for including additional [PSBT][topic psbt] fields
  that provide signing devices with contract commitment data required to participate in
  [Pay-to-Contract][topic p2c] protocols (see [Newsletter #184][news184 psbt]).

> - [BIPs #1364][] makes adds additional detail to the text for the

Reviewer 2: ```suggestion
- [BIPs #1364][] adds additional detail to the text for the
```
